core/src/main/scala/org/apache/spark/api/python/PythonRunner.scala

```diff
@@ -845,9 +843,7 @@ private[spark] class PythonRunner(
         try {
           stream.readInt() match {
             case length if length > 0 =>
-              val obj = new Array[Byte](length)
-              stream.readFully(obj)
-              obj
+              PythonWorkerUtils.readBytes(length, stream)
             case 0 => Array.emptyByteArray
```

Comment:
Can we use `PythonWorkerUtils.readBytes` if we have `length=0` or we need to handle that separately?

Reply:
That's a good point. I think we can include this in `readBytes` to always avoid creating a new empty array.